src/executables/sysrepocfg.c

```diff
@@ -0,0 +1,775 @@
+/*
+ * @file sysrepocfg.c
+ * @author Rastislav Szabo <raszabo@cisco.com>, Lukas Macko <lmacko@cisco.com>,
+ *         Milan Lenco <milan.lenco@pantheon.tech>
+ * @brief Sysrepo configuration tool (sysrepocfg) implementation.
+ *
+ * @copyright
+ * Copyright 2016 Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <dirent.h>
+#include <assert.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <ctype.h>
+#include <libyang/libyang.h>
+
+#include "sr_common.h"
+#include "client_library.h"
+
+#define EXPECTED_MAX_INPUT_FILE_SIZE  4096
+#define DEFAULT_TEXT_EDITOR           "vi"
```

Comment:
When no editor is specified via env. variable or an argument, print an error message suggesting how it could be specified rather than opening in default editor.